2. Create a data frame with three columns- Name, Age and Rating
a. Find the sum of columns
b. Find average age and rating
c. Find the standard deviation
d. Describe the data frame

In [2]:
import pandas as pd
#a.
data={
    'Name':['Rakim','Rahul','Rohan','Ram'],
    'Age': [25,60,40,55],
    'Rating':[6.3,8.5,8.9,4.5]
}
df=pd.DataFrame(data)
Sum=df['Age'].sum()
print("Sum of age:",Sum)
Avg=df['Age'].mean()
print("Average of age:",Avg)
Sd=df['Age'].std()
print("Standard deviation of age:",Sd)
print("Description\n",df.describe().round(2)) #Round 2 will set 2 digits after Decimal


Sum of age: 180
Average of age: 45.0
Standard deviation of age: 15.811388300841896
Description
          Age  Rating
count   4.00    4.00
mean   45.00    7.05
std    15.81    2.05
min    25.00    4.50
25%    36.25    5.85
50%    47.50    7.40
75%    56.25    8.60
max    60.00    8.90


The output of `df.describe()` provides a statistical summary of the numerical columns in the DataFrame. Here’s a breakdown of each row in the summary:

### Age Column
- **Count**: There are 4 values in the "Age" column.
- **Mean**: The average age is 45 years.
- **Standard Deviation (std)**: The age values have a standard deviation of 15.81. This means the ages are somewhat spread out from the mean.
- **Min**: The minimum age is 25.
- **25% (1st Quartile)**: 36.25 is the age below which 25% of the data falls.
- **50% (Median)**: 47.5 is the middle value (median) for age, meaning half the values are below this and half are above.
- **75% (3rd Quartile)**: 56.25 is the age below which 75% of the data falls.
- **Max**: The maximum age is 60.

### Rating Column
- **Count**: There are 4 values in the "Rating" column.
- **Mean**: The average rating is 7.05.
- **Standard Deviation (std)**: The ratings have a standard deviation of 2.05, indicating a moderate spread around the mean.
- **Min**: The lowest rating is 4.5.
- **25% (1st Quartile)**: 5.85 is the rating below which 25% of the data falls.
- **50% (Median)**: 7.4 is the middle value (median) for ratings.
- **75% (3rd Quartile)**: 8.6 is the rating below which 75% of the data falls.
- **Max**: The highest rating is 8.9.

### Key Takeaways
- **Age**: Ages are moderately spread with a mean age of 45, and the values range from 25 to 60.
- **Rating**: Ratings have a slight spread, with an average of 7.05 and a range from 4.5 to 8.9.